### PR TITLE
Remove buildmode=plugin

### PR DESCRIPTION
**Pull request by cgrinds**
Add support for cluster simulator
WIP Implement Caddy style plugins for collectors
Fix go vet warnings in node.go

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -102,44 +100,6 @@ harvest: deps
 	@echo "Building grafana tool"
 	@GOOS=$(GOOS) GOARCH=$(GOARCH) go build -o bin/grafana -ldflags=$(LD_FLAGS) cmd/tools/grafana/main/main.go
 
-###############################################################################
-# Collectors
-###############################################################################
-collectors: deps
```
Comment (rahulguptajss) [inline Makefile:108]: Thank you so much for removing for loops 🥇

File: cmd/collectors/zapi/collector/zapi.go [hunk 2]
```diff
@@ -103,6 +113,25 @@ func (me *Zapi) InitVars() error {
 	return nil
 }
 
+func (me *Zapi) LoadPlugin(kind string, abc *plugin.AbstractPlugin) plugin.Plugin {
+	switch kind {
```
Comment (rahulguptajss) [inline cmd/collectors/zapi/collector/zapi.go:117]: may be log for default case in switch?

File: pkg/api/ontapi/zapi/client.go [hunk 3]
```diff
@@ -66,8 +64,11 @@ func New(config *node.Node) (*Client, error) {
 		return nil, errors.New(errors.MISSING_PARAM, "addr")
 	}
 
-	url = "https://" + addr + ":443/servlets/netapp.servlets.admin.XMLrequest_filer"
-
+	if config.GetChildContentS("is_kfs") == "true" {
```
Comment (rahulguptajss) [inline pkg/api/ontapi/zapi/client.go:67]: do you want to add this variable to static structs in harvest doctor?

File: cmd/poller/poller.go [hunk 4]
```diff
@@ -645,20 +654,16 @@ func (me *Poller) load_exporter(name string) exporter.Exporter {
 		return nil
 	}
 
-	binpath = path.Join(me.options.HomePath, "bin", "exporters")
-
-	if sym, err = dload.LoadFuncFromModule(binpath, strings.ToLower(class), "New"); err != nil {
-		logger.Error().Msgf("dload: %v", err.Error())
+	absExp := exporter.New(class, name, me.options, params)
```
Comment (rahulguptajss) [inline cmd/poller/poller.go:657]: where did we initialized class var in this method?

File: cmd/poller/poller.go [hunk 5]
```diff
@@ -570,8 +557,11 @@ func (me *Poller) load_collector(class, object string) error {
 				continue
 			}
 
-			col = NewFunc(collector.New(class, object.GetNameS(), me.options, template.Copy()))
-			if err = col.Init(); err != nil {
+			col, err = me.newCollector(class, object.GetNameS(), template)
+			if col == nil {
+				continue
```
Comment (rahulguptajss) [inline cmd/poller/poller.go:562]: add log if it is null?

File: cmd/poller/poller.go [hunk 6]
```diff
@@ -590,17 +580,19 @@ func (me *Poller) load_collector(class, object string) error {
 	logger.Debug().Msgf("initialized (%s) with %d objects", class, len(collectors))
 	// link each collector with requested exporter & update metadata
 	for _, col = range collectors {
-
+		if col == nil {
+			continue
```
Comment (rahulguptajss) [inline cmd/poller/poller.go:584]: add log if null?

File: cmd/poller/collector/collector.go [hunk 7]
```diff
@@ -95,6 +94,13 @@ type AbstractCollector struct {
 	countMux *sync.Mutex // used for atomic access to collectCount
 }
 
+func (AbstractCollector) HarvestModule() plugin.ModuleInfo {
```
Comment (rahulguptajss) [inline cmd/poller/collector/collector.go:97]: looks like a placeholder to make abtractcollector interface member. is Unix a good default?

File: cmd/collectors/zapi/collector/zapi.go (hunk 2 above)
Comment (cgrinds) [reply cmd/collectors/zapi/collector/zapi.go:117]: done - this  method will eventually be rewritten using modules too

File: cmd/poller/poller.go (hunk 4 above)
Comment (cgrinds) [reply cmd/poller/poller.go:657]: line 652
if class = params.GetChildContentS("exporter"); class == "" {

File: cmd/poller/collector/collector.go (hunk 7 above)
Comment (cgrinds) [reply cmd/poller/collector/collector.go:97]: Good catch - this is a placeholder that can be removed since the pattern is shown in pollers

**Review by rahulguptajss — APPROVED** [review]
looks good. One general comment for constants. Instead of hardcoded strings, we should move them constant package in pkg folder

**Pull request merged**